raco/backends/myria/myria.py

```diff
@@ -1721,6 +1824,193 @@ def fire(self, op):
+                    expr.input = child
+                elif isinstance(expr, algebra.BinaryOperator):
+                    if idx == 0:
+                        expr.left = child
+                    else:
+                        expr.right = child
+                elif isinstance(expr, algebra.NaryOperator):
+                    expr.args[idx] = child
+        return expr
+
+
+class RemoveSingleSplit(rules.Rule):
+
+    def fire(self, op):
+        if not isinstance(op, algebra.UntilConvergence):
+            return op
+
+        parent_map = {}
+        op.collectParents(parent_map)
+        for idx, child in enumerate(op.children()):
+            if not isinstance(child, MyriaSplitProducer):
+                continue
+            if len(child.consumers) != 1:
+                continue
+            producer = parent_map[id(child.consumers[0])][0]
+            if not isinstance(producer, MyriaShuffleProducer):
+                continue
+            producer.input = child.input
+            consumer = parent_map[id(producer)][0]
+            upper_op = parent_map[id(consumer)][0]
+            new_consumer = MyriaConsumer(None, producer)
+            if isinstance(upper_op, algebra.UnaryOperator):
+                upper_op.input = new_consumer
+            elif isinstance(upper_op, algebra.BinaryOperator):
+                if upper_op.left == parent_map[id(producer)][0]:
+                    upper_op.left = new_consumer
+                else:
+                    upper_op.right = new_consumer
+            else:
+                for idx, t in enumerate(upper_op.args):
+                    if upper_op.args[idx] == parent_map[id(producer)][0]:
+                        upper_op.args[idx] = new_consumer
+                        break
+            op.args[idx] = producer
+        return op
+
+
+class DoUntilConvergence(rules.Rule):
+
+    @staticmethod
+    def replace_scan_with_idb_consumer(expr, idb_controllers, idb_producers):
+        if isinstance(expr, algebra.ZeroaryOperator):
+            return
+        for idx, op in enumerate(expr.children()):
+            if isinstance(op, algebra.ScanIDB):
+                if op.name not in idb_producers:
+                    idb_producers[op.name] = \
+                        MyriaSplitProducer(idb_controllers[op.name])
+                consumer = MyriaConsumer(None, idb_producers[op.name])
+                if isinstance(expr, algebra.UnaryOperator):
+                    expr.input = consumer
+                elif isinstance(expr, algebra.BinaryOperator):
+                    if idx == 0:
+                        expr.left = consumer
+                    else:
+                        expr.right = consumer
+                elif isinstance(expr, algebra.NaryOperator):
+                    expr.args[idx] = consumer
+                if isinstance(expr, algebra.Select) and \
+                        isinstance(expr.condition, expression.boolean.GTEQ):
```

Comment:
It is, but the backend can't handle it yet. Since it's on top of `count` which generates integers, `GTEQ` and `GT` should be equivalent. (I need to add it to the backend though)